### ISSUE TITLE
Build errors (incompatible pointer types)

### ISSUE THREAD

**Issue by miroirdelame** [kaltura/nginx-vod-module#1363]
Hello,

I've been building nginx with my custom set of modules for a while, and that always worked.
I tried again today, and I got compiling errors (see attached [build-errors.txt](https://github.com/kaltura/nginx-vod-module/files/8967466/build-errors.txt)), always about incompatible pointer types.

I've also attached the script ([nginx-rebuild.txt](https://github.com/kaltura/nginx-vod-module/files/8967470/nginx-rebuild.txt)) that I use and that always worked until today.

Thanks in advance

**Comment by jessp01** [comment]
Hi @miroirdelame ,

What GCC version are you using (and is it different to the one you used last time?)

**Comment by erankor** [comment]
Yeah, I knew was this was coming... nginx changed the internal API in this commit - https://github.com/nginx/nginx/commit/3aef1d693f3cc431563a7e6a6aba6a34e5290f03
I'll try to get to it next week...
Btw, if you're building for production use, IMHO working against master is not recommended. General good practice is to use explicit versions for all components, so that you are in control of what goes inside the build.

**Comment by miroirdelame** [comment]
@jessp01 : Nginx 1.23.0 and gcc 9.4.0

@erankor : Thank you for your work, i'll wait patiently.
Btw, the only "master" stuff are the contributed modules, nginx is the latest stable version in the Nginx Ubuntu repo (mainline).
I test my build on a testing environment before sending it to prod. So my prod will remain unchanged until your commit.

**Comment by erankor** [comment]
Pushed support for nginx 1.23.0 to master + tagged 1.30
